framework/logging.cpp

```diff
@@ -1547,6 +1616,24 @@ static void print_content_single_line(int fd, std::string_view before,
     writeln(fd, before, escape_for_single_line(message, escaped), after);
 }
 
+static std::string_view get_skip_message(int thread_num, std::string &skip_message){
+    auto log = log_for_thread(thread_num);
+    struct mmap_region r = mmap_file(log.log_fd);
+    auto ptr = static_cast<const char *>(r.base);
+    const char *end = ptr + r.size;
+    const char *delim;
+
+    for ( ; ptr < end && (delim = strnchr(ptr, '\0', end - ptr)) != NULL; ptr = delim + 1) {
+        ptr++;
+        if(*ptr == 'S'){
+            skip_message.assign(ptr+3, delim-(ptr+3));
```

Comment:
This may go beyond `end`.

Reply:
Can you explain more on this? Cannot think of a scenario.

I thought the `delim` will ensure it doesn't go over the end. For e.g. message can be only of this format

`I> This is info\0W> This is warning\0E> This is error\0D> This is debug\0S> This is skip\0`

Reply:
In theory it could. If the message is less than 3 chars (e.g. writing ended abruptly due to a crash), it only has `ptr < end` as the loop condition. Than the expression `ptr + 3` will go beyond.

Reply:
This has been handled in the second commit which i have pushed.